File: 2025-09-18 - Vetores/Exemplo_C#/VetorInterface.cs
// ------------------------------------------------------------------------
// VetorInterface.cs
// Interface que declara os métodos em camelCase
// ------------------------------------------------------------------------
public interface VetorInterface
{
   bool inserir(Node item);
   bool inserirEm(Node item, int pos);
   bool removerPorCodigo(string codigo);
   bool removerPorPosicao(int pos);
   Node? buscarPorCodigo(string codigo);
   bool atualizarPorCodigo(string codigo, Node novo);
   void imprimirVetor();
   int tamanho();
   bool estaVazio();
}

File: 2025-09-18 - Vetores/Exemplo_Java/VetorInterface.java
// ------------------------------------------------------------------------
// VetorInterface.java
// Interface que declara os métodos públicos esperados para a estrutura Vetor
// ------------------------------------------------------------------------
public interface VetorInterface {
   boolean inserir(Node item);
   boolean inserirEm(Node item, int pos);
   boolean removerPorCodigo(String codigo);
   boolean removerPorPosicao(int pos);
   Node buscarPorCodigo(String codigo);
   boolean atualizarPorCodigo(String codigo, Node novo);
   void imprimirVetor();
   int tamanho();
   boolean estaVazio();
}
